src/gluonts/dataset/loader.py

```diff
@@ -203,22 +203,23 @@ def TrainDataLoader(
         dataset = PseudoShuffled(dataset, shuffle_buffer_length)
 
     transform += Batch(batch_size=batch_size) + AdhocTransform(stack_fn)
-    batches: DataLoader = transform.apply(dataset, is_train=True)
+    transformed_dataset = transform.apply(dataset, is_train=True)
 
     if num_workers is not None:
         batches = MultiProcessLoader(
-            batches,
+            transformed_dataset,
             decode_fn=decode_fn,
             num_workers=num_workers,
             max_queue_size=num_prefetch,
         )
-
-    batches = iter(batches)
+        iterator = iter(batches)
```

Comment:
You don't like renaming?

Reply:
Most tools discourage shadowing ... for mypy you need to explicitly set that you are allowed to change the type (which GluonTS probably does)

Reply:
Right, because it makes more complicated for them. Personally, I think it's still fine.